src/main/java/com/nha/abdm/wrapper/hrp/serviceImpl/LogsTableService.java

```diff
@@ -124,11 +165,37 @@ public List<CareContextBuilder> getSelectedCareContexts(String linkRefNumber, Li
 
         return null;
     }
-    public String getStatus(JsonNode data) {
-        RequestLogs existingRecord=logsRepo.findByClientRequestId(data.get("requestId").asText());
+    public String getStatus(ResponseHelper data) {
+        RequestLogs existingRecord=logsRepo.findByClientRequestId(data.getRequestId());
         if(existingRecord!=null){
             return existingRecord.getResponse().toString();
         }
         return "Record failed but stored in database";
     }
+        public void setStatus(OnAddCareContextResponse data) {
+        RequestLogs existingRecord=logsRepo.findByGatewayRequestId2(data.getResp().getRequestId());
+        try{
+        if(existingRecord!=null){
+            Query query = new Query(Criteria.where("gatewayRequestId2").is(data.getResp().getRequestId()));
+            Update update = (new Update()).set("response", data.getAcknowledgement().getStatus());
+            mongoTemplate.updateFirst(query, update, RequestLogs.class);
+            LinkRecordsResponse linkRecordsResponse=(LinkRecordsResponse) existingRecord.getRawResponse().get("LinkRecordsResponse");
+            patientTableService.updateCareContextStatus(linkRecordsResponse.getPatientReference(),linkRecordsResponse.getPatient().getCareContexts());
```

Comment:
Shouldn't the care contexts for Patients be updated only for Successful Acknowledgement?

Reply:
For example, if the hip linking doesn't have an ABHA address to link, the careContexts will be saved in the wrapper, such that whenever there is a discover call, the patient can fetch them.